test/functional/scram_sha_256_tests.js

```diff
@@ -0,0 +1,216 @@
+'use strict';
+
+const expect = require('chai').expect;
+const sinon = require('sinon');
+const ScramSHA256 = require('mongodb-core').ScramSHA256;
+const MongoError = require('mongodb-core').MongoError;
+const setupDatabase = require('./shared').setupDatabase;
+const withClient = require('./shared').withClient;
+const MongoClient = require('../../lib/mongo_client');
+
+describe('SCRAM-SHA-256 auth', function() {
+  const test = {};
+  const userMap = {
+    sha1: {
+      description: 'user with sha1 credentials',
+      username: 'sha1',
+      password: 'sha1',
+      mechanisms: ['SCRAM-SHA-1']
+    },
+    sha256: {
+      description: 'user with sha256 credentials',
+      username: 'sha256',
+      password: 'sha256',
+      mechanisms: ['SCRAM-SHA-256']
+    },
+    both: {
+      description: 'user with both credentials',
+      username: 'both',
+      password: 'both',
+      mechanisms: ['SCRAM-SHA-1', 'SCRAM-SHA-256']
+    }
+  };
+
+  function makeConnectionString(config, username, password) {
```

Comment:
Can we improve [config.url](https://github.com/mongodb/node-mongodb-native/blob/master/test/config.js#L74-L80) to support this use case?

Reply:
I'd be on board with that, though idk if it should be part of this change.

Reply:
okay cool